### PR TITLE
Drop IOVEC_NULL

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/basic/iovec-util.h [hunk 1]
```diff
@@ -12,16 +12,14 @@ size_t iovec_total_size(const struct iovec *iovec, size_t n);
...
-
 #define IOVEC_MAKE(base, len) (struct iovec) { .iov_base = (base), .iov_len = (len) }
 #define IOVEC_MAKE_STRING(string)                       \
         ({                                              \
                 const char *_s = (string);              \
                 IOVEC_MAKE((char*) _s, strlen(_s));     \
         })
 
-#define TAKE_IOVEC(p) TAKE_GENERIC((p), struct iovec, IOVEC_NULL)
+#define TAKE_IOVEC(p) TAKE_GENERIC((p), struct iovec, {})
```
Comment (YHNdnzj) [inline src/basic/iovec-util.h]: This is covered by the plain `TAKE_STRUCT`. No need for a dedicated helper.

File: src/basic/iovec-util.c [hunk 2]
```diff
@@ -62,12 +62,12 @@ char* set_iovec_string_field_free(struct iovec *iovec, size_t *n_iovec, const ch
         return x;
 }
 
-void iovec_array_free(struct iovec *iov, size_t n) {
-        if (!iov)
+void iovec_array_free(struct iovec *iovec, size_t n) {
+        if (!iovec)
                 return;
 
         for (size_t i = 0; i < n; i++)
```
Comment (YHNdnzj) [inline src/basic/iovec-util.c]: While at it, maybe use `FOREACH_ARRAY` or `free_many` instead?

File: src/basic/iovec-util.c [hunk 3]
```diff
@@ -62,12 +62,12 @@ char* set_iovec_string_field_free(struct iovec *iovec, size_t *n_iovec, const ch
         return x;
 }
 
-void iovec_array_free(struct iovec *iov, size_t n) {
-        if (!iov)
+void iovec_array_free(struct iovec *iovec, size_t n) {
```
Comment (YHNdnzj) [inline src/basic/iovec-util.c:65]: Nowadays we make such destructors return NULL instead of void.

Comment (keszybz) [reply src/basic/iovec-util.c:65]: It's used in CLEANUP_ARRAY, which wants a function that returns `void`.

File: src/basic/iovec-util.c (hunk 2 above)
Comment (keszybz) [reply src/basic/iovec-util.c]: I switched it to FOREACH_ARRAY.

**Review by yuwata — APPROVED** [review]

**Pull request merged**